### ISSUE TITLE
Static link problem

### ISSUE THREAD

**Issue by abeq** [plv8/plv8#163]
Run 'make static' and then psql -c 'CREATE EXTENSION plv8' but following error has occurred :
ERROR: could not load library "/usr/pgsql-9.4/lib/plv8.so": libv8.so: cannot open shared object file: No such file or directory
Why does plv8 with v8 static link needs libv8.so, shared library?

**Comment by JerrySievert** [comment]
can you share the os version and which version of plv8 you are compiling against? (master? a tag?)

i don't see the issue occurring under OS X or ubuntu, so would love some details to be able to reproduce it.

thanks!

**Comment by abeq** [comment]
Hi, my environment is as followings:
CentOS 7.2.1511 64-bit on Vegrant/Parallel in Mac
plv8: current
v8: 4.10.253 (as Makefile.v8)
If you need further informations, please let me know.

**Comment by abeq** [comment]
And I tried I tried followings procedure:
1. Build v8 with "make x64.relaese library=shared strictaliasing=off console=readline".
2. Copy out/x64.release/lib*.so to /usr/lib64/.
3. Export V8_SRCDIR and V8_OUTDIR.
4. Run make and make install plv8.
But when run psql -c 'CREATE EXTENSION plv8' with postgres user, following error has occurred:
ERROR: could not load library "usr/pgsql-9.4/lib/plv8.so": /usr/pgsql-9.4/lib/plv8.so: undefined symbol: _ZN2v88platform21CreateDefaultPlatformEi

**Comment by JerrySievert** [comment]
i can look to try to reproduce this, but it looks like libplatform isn't being built out.  can you try with snapshot=off in your compile?

also, what version of g++ are you compiling with?

thanks!

**Comment by JerrySievert** [comment]
also, one thing to note, _master_ is the development branch, if you want something considered stable, please use one of the releases (which can be downloaded via the releases link, or accessed as a tag).

1.4.4 would be considered stable against older (3.x) versions of v8, 1.5.0 would be considered stable against older (4.0-4.4) versions of v8.

_master_ is the development branch for newer (4.5-4.10) versions of v8, and will eventually be released as plv8 2.0.0.

**Comment by JerrySievert** [comment]
@abeq i installed centos 7 locally, did a `git clone`, and `make static`.

```
[jerry@centos plv8]$ make install
/usr/bin/mkdir -p '/usr/local/pgsql/lib'
/usr/bin/mkdir -p '/usr/local/pgsql/share/extension'
/usr/bin/mkdir -p '/usr/local/pgsql/share/extension'
/usr/bin/install -c -m 755  plv8.so '/usr/local/pgsql/lib/plv8.so'
/usr/bin/install -c -m 644 .//plv8.control '/usr/local/pgsql/share/extension/'
/usr/bin/install -c -m 644 .//plv8.control .//plv8--2.0.0-dev.sql .//plcoffee.control .//plcoffee--2.0.0-dev.sql .//plls.control .//plls--2.0.0-dev.sql  '/usr/local/pgsql/share/extension/'
[jerry@centos plv8]$ make installcheck
/usr/local/pgsql/lib/pgxs/src/makefiles/../../src/test/regress/pg_regress --inputdir=./ --bindir='/usr/local/pgsql/bin'    --dbname=contrib_regression init-extension plv8 plv8-errors inline json startup_pre startup varparam json_conv jsonb_conv window guc es6 arraybuffer composites dialect
(using postmaster on Unix socket, default port)
============== dropping database "contrib_regression" ==============
NOTICE:  database "contrib_regression" does not exist, skipping
DROP DATABASE
============== creating database "contrib_regression" ==============
CREATE DATABASE
ALTER DATABASE
============== running regression test queries        ==============
test init-extension           ... ok
test plv8                     ... ok
test plv8-errors              ... ok
test inline                   ... ok
test json                     ... ok
test startup_pre              ... ok
test startup                  ... ok
test varparam                 ... ok
test json_conv                ... ok
test jsonb_conv               ... ok
test window                   ... ok
test guc                      ... ok
test es6                      ... ok
test arraybuffer              ... ok
test composites               ... ok
test dialect                  ... ok

======================
 All 16 tests passed.
======================
```

not sure what is going on with your environment - how can i help further?
